apps/web/src/stores/latte.ts

```diff
@@ -0,0 +1,215 @@
+    }),
+
+  setError: (error: string | undefined) => set({ error }),
+
+  // Changes actions
+  setChanges: (
+    changes: LatteChange[] | ((prev: LatteChange[]) => LatteChange[]),
+  ) =>
+    set((state) => ({
+      changes: typeof changes === 'function' ? changes(state.changes) : changes,
+    })),
+
+  addChange: (change: LatteChange) =>
+    set((state) => ({
+      changes: [...state.changes, change],
+    })),
+
+  updateChange: (
+    draftUuid: string,
+    documentUuid: string,
+    updater: (change: LatteChange) => LatteChange,
+  ) =>
+    set((state) => {
+      const index = state.changes.findIndex(
+        (change) =>
+          change.draftUuid === draftUuid &&
+          change.current.documentUuid === documentUuid,
+      )
+
+      if (index === -1) return state
+
+      const updatedChanges = [...state.changes]
+      updatedChanges[index] = updater(updatedChanges[index]!)
+
+      return { changes: updatedChanges }
+    }),
+
+  removeChange: (draftUuid: string, documentUuid: string) =>
+    set((state) => ({
+      changes: state.changes.filter(
+        (change) =>
+          !(
+            change.draftUuid === draftUuid &&
+            change.current.documentUuid === documentUuid
+          ),
+      ),
+    })),
+
+  setLatteActionsFeedbackUuid: (uuid: string | undefined) =>
+    set({ latteActionsFeedbackUuid: uuid }),
+
+  // Reset functions
+  resetChat: () =>
+    set({
+      isLoading: false,
+      interactions: [],
+      error: undefined,
+      threadUuid: undefined,
+    }),
+
+  resetChanges: () =>
+    set({
+      changes: [],
+      latteActionsFeedbackUuid: undefined,
+    }),
+
+  resetAll: () =>
+    set({
+      isLoading: false,
+      interactions: [],
+      error: undefined,
+      changes: [],
+      latteActionsFeedbackUuid: undefined,
+      threadUuid: undefined,
+    }),
+}))
+
+export const useLatteStore = () => {
+  const store = useStore()
+  const searchParams = useSearchParams()
+  const replaceUrlState = useCallback(
+    (uuid: string | undefined) => {
+      const newSearchParams = new URLSearchParams(searchParams.toString())
+      if (uuid) {
+        newSearchParams.set('latteThreadUuid', uuid)
+      } else {
+        newSearchParams.delete('latteThreadUuid')
+      }
+
+      const newUrl = `?${newSearchParams.toString()}`
+      const currentUrl = `?${searchParams.toString()}`
+
+      if (newUrl !== currentUrl) {
+        window.history.replaceState(null, '', newUrl)
+      }
+    },
+    [searchParams],
+  )
+  const setThreadUuid = useCallback(
+    (uuid: string | undefined) => {
+      store.setThreadUuid(uuid)
+      replaceUrlState(uuid)
+    },
+    [store, replaceUrlState],
+  )
+  const resetChat = useCallback(() => {
+    store.resetChat()
+    replaceUrlState(undefined)
+  }, [store, replaceUrlState])
+
+  return useMemo(
+    () => ({
+      ...store,
+      setThreadUuid,
+      resetChat,
+    }),
+    [store, setThreadUuid, resetChat],
+  )
+}
```

Comment:
the same state that previously live in useLatte but in a zustand global store and with proper syncing with the url's search params